### PR TITLE
Context Parameters

### PR DESCRIPTION
**Pull request by jlacivita**

### CODE REVIEW

File: src/js/shared/Events/index.mjs [hunk 1]
```diff
@@ -64,7 +72,9 @@ const doListen = function(module, event, callback, once) {
 
     let events = (event === '*' ? validEvents[module] : [event]) // explodes wildcards into an array
     let promises = []
-    const key = module + '.' + event // this might be a wildcard, e.g. 'lifecycle.*'
+    const hasContext = Object.values(context).length > 0
+    const contextKey = Object.keys(context).sort().map(key => key + '=' + JSON.stringify(context[key])).join('&')
```
Comment (kpears201) [inline src/js/shared/Events/index.mjs:76]: I don't quite understand how this works.
The callback is stored now using the contextKey, but when emitting the event, I don't think it uses this same key to find the callback?

File: src/js/shared/Prop/index.mjs [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
-  if (args.length === 0) {
+function prop(moduleName, key, params, callbackOrValue, immutable, readonly, contextParameterCount) {
+  const numArgs = Object.values(params).length
+
+  if (numArgs === contextParameterCount && !callbackOrValue) {
     // getter
-    return Transport.send(moduleName, key)
-  } else if (args.length === 1 && typeof args[0] === 'function') {
+    return Transport.send(moduleName, key, )
+  } else if (numArgs === (contextParameterCount) && typeof callbackOrValue === 'function') {
```
Comment (kpears201) [inline src/js/shared/Prop/index.mjs:10]: these parenthesis are unnecessary here

File: src/js/shared/Events/index.mjs (hunk 1 above)
Comment (jlacivita) [reply src/js/shared/Events/index.mjs:76]: You might be right. let me look closer at my tests / code.

File: src/js/shared/Prop/index.mjs (hunk 2 above)
Comment (jlacivita) [reply src/js/shared/Prop/index.mjs:10]: will fix.

File: src/js/shared/Events/index.mjs (hunk 1 above)
Comment (jlacivita) [reply src/js/shared/Events/index.mjs:136]: Yeah, it was totally broken. I had to refactor the Event/Transport integration, but it's much more solid now

**Pull request merged**